app/code/local/Dwyera/Pinpay/Model/PaymentMethod.php

```diff
@@ -38,16 +39,63 @@ public function assignData($data)
             $data = new Varien_Object($data);
         }
 
+        //Check if customer tokenization is enabled.
+        $block = new Dwyera_Pinpay_Block_Form;
+        $customerTokenizationEnabled = $block->iscustomerTokenizationEnabled();
+        $customerToken = $data->getCustomerToken();
+
+        //To Check if the customer is logged in.
+        $customerData = $this->getCustomer();
+
+        //Get the card Token from data
         $cardToken = $data->getCardToken();
+
+        //Check if customer token is enabled, customer is logged in and the card token is not the same as the one in database.
+        /* Comparing the card token with existing token to make sure the following things donot happen
+        1. The unnecessary API calls to the Pinpayment API is avoided if the card token is same
+         */
+        if($customerTokenizationEnabled && $customerData != false && $customerData->getData('pinpayment_card_token') != $cardToken){
+            if ($data->getData('card_action') == "save_card" &&  $customerToken =="") {
+                $customerTokenDetails = $this->getCustomerToken($data);
+                $customerToken = $customerTokenDetails->getcustomerToken();
+                if ($customerToken != "") {
+                    //For first time save all details like customer token, display number and card token to database.
+                    $customerData->setData('pinpayment_customer_token', $customerToken);
+                    $customerData->setData('pinpayment_card_display_number', $customerTokenDetails->getPrimaryCardDisplayNumber());
+                    $customerData->setData('pinpayment_card_token', $customerTokenDetails->getCardToken());
+                    $customerData->save();
```

Comment:
@zainengineer Customer can save only one card at any time. If customer saves new card, the old card will be removed. At all times there would be just 1 card saved in customer token.